core/drivers/nvmem/nvmem_huk.c

```diff
@@ -31,6 +31,8 @@ TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
 
 	memcpy(hwkey->data, huk, HW_UNIQUE_KEY_LENGTH);
 
+	return res;
+
 out_free_cell:
 	nvmem_put_cell(huk_cell);
```

Comment:
With the change you propose, I think you should not "put" the NVMEM cell even if line 23 fails.
The test on the NVMEM cell regarding `HW_UNIQUE_KEY_LENGTH` should rather be done from `nvmem_huk_probe()` (or `nvmem_huk_get_cell()` if you prefer) and `tee_otp_get_hw_unique_key()` to return an error code if `huk_cell` is NULL.

As `tee_otp_get_hw_unique_key()` is likely to be called during OP-TEE initialization, I think you could even get the cell data straight from `nvmem_huk_probe()` so that if probe succeeds, `tee_otp_get_hw_unique_key()` will always succeed also.